instructions/steps.md

```diff
@@ -1023,488 +1025,62 @@ This is a slice notation.
-  {}
-
-```
-
-Dafny uses the fact that the string `ParseAwsKmsIdentifier` will succeed
-with the fact that this string starts with `arn:`
-to know that `ParseAwsKmsArn` MUST succeed.
-We can then wrap this `arn` so that we can compare the two calls
-as the specification requires.
-
-Looking at our specification,
-our next two requirements are very similar.
-They differ in how the string should start
-and if `MultiRegionAwsKmsIdentifier?` should return true or false
-
-<!-- !test check MultiRegionAwsKmsIdentifier?Correct other requirements -->
-```dafny
-
-    //= aws-kms-key-arn.txt#2.9
-    //= type=implication
-    //# If the input starts with "alias/", this an AWS KMS alias and
-    //# not a multi-Region key id and MUST return false.
-    ensures "alias/" <= identifier
-      ==>
-        var resource := ParseAwsKmsIdentifier(identifier).value;
-        !MultiRegionAwsKmsIdentifier?(ParseAwsKmsIdentifier(identifier).value)
-
-    //= aws-kms-key-arn.txt#2.9
-    //= type=implication
-    //# If the input starts
-    //# with "mrk-", this is a multi-Region key id and MUST return true.
-    ensures "mrk-" <= identifier
-      ==>
-        var resource := ParseAwsKmsIdentifier(identifier).value;
-        MultiRegionAwsKmsIdentifier?(resource)
-
-```
-
-Finally the specification says
-"If the input does not start with any of the above,
-this is not a multi-Region key id and MUST return false."
-This is just a negation of the three starts with we already have.
-
-There are a few ways to express this,
-but this one mirrors the specification to make it easier
-to see the correspondence.
-
-<!-- !test check MultiRegionAwsKmsIdentifier?Correct final ensures -->
-```dafny
-
-    //= aws-kms-key-arn.txt#2.9
-    //= type=implication
-    //# If
-    //# the input does not start with any of the above, this is not a multi-
-    //# Region key id and MUST return false.
-    ensures
-        && !("arn:" <= identifier )
-        && !("alias/" <= identifier )
-        && !("mrk-" <= identifier )
-      ==>
-        var resource := ParseAwsKmsIdentifier(identifier);
-        !MultiRegionAwsKmsIdentifier?(resource.value)
+Let's run Duvet to see where we are.
+It will fail because we have not completed all sections.
 
+```bash
+make duvet_report
 ```
 
-Putting that all together we get.
+You have completed section 2.5!
+Now we know that every requirement in that section is annotated
+and we know that our implementation follows it's Dafny specification.
 
-<!-- !test check MultiRegionAwsKmsIdentifier?Correct complete -->
-```dafny
+Before we move on,
+remember that we could put `duvet` into our CI build process.
+This would create a mechanism to ensure that every requirement is meet.
+It also is a way to track changes
+if new requirements are added the project.
 
-  lemma MultiRegionAwsKmsIdentifier?Correct(identifier: string)
-    requires ParseAwsKmsIdentifier(identifier).Success?
+Why don't we execute it?!
```

Comment:
Expand "it", sounds like you mean `duvet` rather than the Dafny code.

I would also have folks look at `Index.dfy` to understand better what they are actually executing.

Reply:
done